File: university-frontend/src/app/core/services/timeslot.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { TimeSlot, Schedule } from '../models/timeslot.model';

@Injectable({
  providedIn: 'root'
})
export class TimeSlotService {
  private http = inject(HttpClient);
  private apiUrl = `${environment.apiUrl}/time-slots`;

  getTimeSlotsByCourseOffering(courseOfferingId: number): Observable<TimeSlot[]> {
    return this.http.get<TimeSlot[]>(`${this.apiUrl}/course-offering/${courseOfferingId}`);
  }

  getStudentSchedule(studentId: number, periodId: number): Observable<Schedule[]> {
    return this.http.get<Schedule[]>(`${this.apiUrl}/student/${studentId}/period/${periodId}`);
  }

  getProfessorSchedule(professorId: number, periodId: number): Observable<Schedule[]> {
    return this.http.get<Schedule[]>(`${this.apiUrl}/professor/${professorId}/period/${periodId}`);
  }

  deleteTimeSlot(timeSlotId: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${timeSlotId}`);
  }
}

File: backend/src/main/java/com/university/application/service/TimeSlotService.java
package com.university.application.service;

import com.university.web.dto.schedule.ScheduleResponseDTO;
import com.university.web.dto.timeslot.TimeSlotResponseDTO;

import java.util.List;

public interface TimeSlotService {
    List<TimeSlotResponseDTO> getTimeSlotsByCourseOffering(Long courseOfferingId);
    List<ScheduleResponseDTO> getStudentSchedule(Long studentId, Long periodId);
    List<ScheduleResponseDTO> getProfessorSchedule(Long professorId, Long periodId);
    void deleteTimeSlot(Long timeSlotId);
}
